### PR TITLE
did the task

### PR DESCRIPTION
**Pull request by vanyachyzh**
[DEMO LINK](https://vanyachyzh.github.io/react_add-todo-form/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,61 +1,134 @@
...
+
+  const addTitle = (e: React.SyntheticEvent) => {
+    const newTitle = (e.target as HTMLInputElement).value;
+
+    setTitleError(false);
+    setTitle(newTitle);
+  };
+
+  const addUser = (e: React.SyntheticEvent) => {
+    const newUser = (e.target as HTMLInputElement).value;
```
Comment (etojeDenys) [inline src/App.tsx:36]:
```suggestion
  const addUser = (e: React.ChangeEvent<HTMLSelectElement>) => {
    const newUser = e.target.value;
```

File: src/App.tsx [hunk 2]
```diff
@@ -1,61 +1,134 @@
...
+
+    if (!user) {
+      setUserError(true);
+
+      return;
+    }
+
+    if (nextUser) {
+      setTodoList(prevState => {
+        return [
```
Comment (etojeDenys) [inline src/App.tsx:61]: can omit
```suggestion
    if (nextUser) {
      setTodoList(prevState =>  [
```

File: src/App.tsx [hunk 3]
```diff
@@ -1,61 +1,134 @@
...
 
 export const App = () => {
+  const [todoList, setTodoList] = useState(todos);
+  const [titleError, setTitleError] = useState(false);
+  const [userError, setUserError] = useState(false);
+  const [user, setUser] = useState('');
+  const [title, setTitle] = useState('');
+
+  const addTitle = (e: React.SyntheticEvent) => {
+    const newTitle = (e.target as HTMLInputElement).value;
```
Comment (etojeDenys) [inline src/App.tsx:29]:
```suggestion
  const addTitle = (e: React.ChangeEvent<HTMLInputElement>) => {
    const newTitle = e.target.value;
```

File: src/components/TodoList/TodoList.tsx [hunk 4]
```diff
@@ -1 +1,17 @@
...
+import { TodoInfo } from '../TodoInfo';
+
+type Props = {
+  todos: Todo[]
+};
+
+export const TodoList: React.FC<Props> = React.memo(({ todos }) => (
+  <ul className="TodoList">
+    {todos.map(todo => {
+      return (
```
Comment (etojeDenys) [inline src/components/TodoList/TodoList.tsx:12]: can  omit

**Review by etojeDenys — CHANGES_REQUESTED** [review]
before creating a todo, check if a user was selected; if not, show an error message next to the select (Please choose a user);
image

File: src/App.tsx [hunk 5]
```diff
@@ -1,61 +1,139 @@
...
+            >
+              Choose a user
+            </option>
+            {usersFromServer.map(u => {
+              return (
+                <option key={u.id}>
+                  {u.name}
+                </option>
+              );
+            })}
```
Comment (VitaliyBondarenko1982) [inline src/App.tsx:124]:
```suggestion
            {usersFromServer.map(u => (
                <option key={u.id}>
                  {u.name}
                </option>
              );
            )}
```

**Review by VitaliyBondarenko1982 — APPROVED** [review]

**Pull request closed without merging**